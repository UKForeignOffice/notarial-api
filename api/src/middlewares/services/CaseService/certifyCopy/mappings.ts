export const order = {
  applicantDetails: "Applicant's details",
  applicantPassport: "Applicant's passport details",
  deliveryAddress: "Delivery address",
  contactDetails: "Contact details",
  parent2ContactDetails: "Parent 2 contact details",
  feedbackConsent: "Feedback consent",
};

export const remap = {
  country: "information.country",
  post: "information.post",
  typeOfDoc: "information.typeOfDocument",
  applicationType: "information.applicationType",
  numCertifiedCopies: "information.numCertifiedCopies",

  phoneNumber: "contactDetails.phoneNumber",
  emailAddress: "contactDetails.emailAddress",

  "parent2Contact.phoneNumber": "parent2ContactDetails.phoneNumber",

  firstName: "applicantDetails.firstName",
  middleName: "applicantDetails.middleName",
  surname: "applicantDetails.surname",

  dateOfBirth: "applicantBirth.dateOfBirth",
  placeOfBirth: "applicantBirth.placeOfBirth",
  countryOfBirth: "applicantBirth.countryOfBirth",

  passportNumber: "applicantPassport.number",
  passportDateOfIssue: "applicantPassport.dateOfIssue",

  childFirstName: "applicantDetails.firstName",
  childMiddleName: "applicantDetails.childMiddleName",
  childSurname: "applicantDetails.childSurname",

  childPassportNumber: "applicantPassport.number",
  childCountryOfBirth: "applicantPassport.countryOfBirth",
  childDateOfBirth: "applicantPassport.dateOfBirth",
  childPlaceOfBirth: "applicantPassport.placeOfBirth",
  childPassportDateOfIssue: "applicantPassport.dateOfIssueOfPassport",

  deliveryAddressLine1: "deliveryAddress.addressLine1",
  deliveryAddressLine2: "deliveryAddress.addressLine2",
  deliveryCity: "deliveryAddress.townOrCity",
  deliveryPostcode: "deliveryAddress.postcodeOrZipCode",
  deliveryCountry: "deliveryAddress.country",

  feedbackConsent: "feedbackConsent.feedbackConsent",
};
