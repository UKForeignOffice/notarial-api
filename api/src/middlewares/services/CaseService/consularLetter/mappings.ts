export const order = {
  applicantDetails: "Applicant's details",
  nokDetails: "Next of kin's details",
  companyDetails: "Funeral director company address",
  deceasedDetails: "Details of the deceased",
  deliveryAddress: "Delivery address",
  contactDetails: "Contact details",
  feedbackConsent: "Feedback consent",
};

export const remap = {
  post: "information.post",
  nextOfKin: "information.applicantIsNextOfKin",

  phoneNumber: "contactDetails.phoneNumber",
  emailAddress: "contactDetails.emailAddress",

  "yourDetails.firstName": "applicantDetails.firstName",
  "yourDetails.middleName": "applicantDetails.middleName",
  "yourDetails.surname": "applicantDetails.surname",
  "yourDetails.relationshipToDeceased": "applicantDetails.applicantsRelationshipToDeceased",
  "yourDetails.relationshipToDeceasedOther": "applicantDetails.applicantsRelationshipToDeceasedIfOther",
  "yourDetails.applicantPassportOrThaiID": "applicantDetails.applicantsPassportOrThaiIdNumber",

  "nextOfKinDetails.firstName": "nokDetails.firstName",
  "nextOfKinDetails.middleName": "nokDetails.middleName",
  "nextOfKinDetails.surname": "nokDetails.surname",
  "nextOfKinDetails.relationshipToDeceased": "nokDetails.applicantsRelationshipToDeceased",
  "nextOfKinDetails.relationshipToDeceasedOther": "nokDetails.applicantsRelationshipToDeceasedIfOther",
  "nextOfKinDetails.applicantPassportOrThaiID": "nokDetails.applicantsPassportOrThaiIdNumber",

  funeralDirectorCompanyName: "companyDetails.companyName",
  funeralDirectorAddressLine1: "companyDetails.addressLine1",
  funeralDirectorAddressLine2: "companyDetails.addressLine2",
  funeralDirectorAddressLine3: "companyDetails.addressLine3",
  funeralDirectorTownOrCity: "companyDetails.townOrCity",
  funeralDirectorPostcode: "companyDetails.postcodeOrZipCode",

  deceasedFirstname: "deceasedDetails.firstName",
  deceasedMiddleName: "deceasedDetails.middleName",
  deceasedSurname: "deceasedDetails.surname",

  addressLine1: "deliveryAddress.addressLine1",
  addressLine2: "deliveryAddress.addressLine2",
  addressLine3: "deliveryAddress.addressLine3",
  townOrCity: "deliveryAddress.townOrCity",
  postcode: "deliveryAddress.postcodeOrZipCode",

  feedbackConsent: "feedbackConsent.feedbackConsent",
};
