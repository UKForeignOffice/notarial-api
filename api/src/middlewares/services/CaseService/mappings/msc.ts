export const order = {
  applicantDetails: "Applicant's details",
  residentStatus: "Applicant's residential status",
  applicantAddress: "Applicant's permanent address",
  applicantLocalAddress: "Applicant's local address",
  applicantPassport: "Applicant's passport",
  maritalStatus: "Marital and civil status",
  marriageDetails: "Marriage",
  deliveryDetails: "Delivery details",
  feedbackConsent: "Feedback consent",
};

export const remap = {
  country: "information.country",
  post: "information.post",
  certifyPassport: "information.certifyPassport",
  feedbackConsent: "feedbackConsent.feedbackConsent",

  dateOfBirth: "applicantDetails.dateOfBirth",
  sex: "applicantDetails.sex",
  emailAddress: "applicantDetails.emailAddress",
  phoneNumber: "applicantDetails.phoneNumber",

  passportDateOfIssue: "applicantPassport.dateOfIssue",
  passportNumber: "applicantPassport.number",

  livesInCountry: "residentStatus.isAResidentOfApplicationCountry",

  addressLine1: "applicantAddress.addressLine1",
  addressLine2: "applicantAddress.addressLine2",
  city: "applicantAddress.city",
  postcode: "applicantAddress.postcode",
  countryAddress: "applicantAddress.country",

  localAddressLine1: "applicantLocalAddress.addressLine1",
  localAddressLine2: "applicantLocalAddress.addressLine2",
  localCity: "applicantLocalAddress.city",
  localPostcode: "applicantLocalAddress.postcode",

  maritalStatus: "maritalStatus.status",

  dateOfMarriage: "marriageDetails.dateOfMarriage",
  placeOfMarriage: "marriageDetails.placeOfMarriage",

  deliveryFirstName: "deliveryDetails.deliveryFirstName",
  deliverySurname: "deliveryDetails.deliverySurname",
  deliveryAddressLine1: "deliveryDetails.deliveryAddressLine1",
  deliveryAddressLine2: "deliveryDetails.deliveryAddressLine2",
  deliveryCity: "deliveryDetails.deliveryCity",
  deliveryPostcode: "deliveryDetails.deliveryPostcode",
  deliveryPhone: "deliveryDetails.deliveryPhone",
};
