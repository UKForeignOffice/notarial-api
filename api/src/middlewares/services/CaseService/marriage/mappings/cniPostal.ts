export const order = {
  applicantDetails: "Applicant's details",
  residentStatus: "Applicant's residential status",
  applicantAddress: "Applicant's permanent address",
  applicantLocalAddress: "Applicant's local address",
  maritalStatus: "Marital and civil status",
  partnerName: "Partner's name",
  partnerAddress: "Partner's address",
  partnerNationality: "Partner's nationality",
  partnerMaritalStatus: "Partner's marital and civil status",
  marriageDetails: "Marriage",
  deliveryDetails: "Delivery details",
  feedbackConsent: "Feedback consent",
};

export const remap = {
  country: "information.country",
  post: "information.post",
  certifyPassport: "information.certifyPassport",
  feedbackConsent: "feedbackConsent.feedbackConsent",

  firstName: "applicantDetails.firstName",
  middleName: "applicantDetails.middleName",
  surname: "applicantDetails.surname",
  emailAddress: "applicantDetails.emailAddress",
  phoneNumber: "applicantDetails.phoneNumber",

  livesInCountry: "residentStatus.isAResidentOfApplicationCountry",

  addressLine1: "applicantAddress.addressLine1",
  addressLine2: "applicantAddress.addressLine2",
  city: "applicantAddress.city",
  postcode: "applicantAddress.postcode",
  countryAddress: "applicantAddress.country",

  localAddressLine1: "applicantLocalAddress.addressLine1",
  localAddressLine2: "applicantLocalAddress.addressLine2",
  localCity: "applicantLocalAddress.city",
  localPostcode: "applicantLocalAddress.postcode",

  maritalStatus: "maritalStatus.status",

  partnerFirstName: "partnerName.partnerFirstName",
  partnerMiddleName: "partnerName.partnerMiddleName",
  partnerSurname: "partnerName.partnerSurname",
  partnerSex: "partnerName.partnerSex",

  partnerAddressLine1: "partnerAddress.partnerAddressLine1",
  partnerAddressLine2: "partnerAddress.partnerAddressLine2",
  partnerAddressCity: "partnerAddress.partnerAddressCity",
  partnerAddressPostcode: "partnerAddress.partnerAddressPostcode",
  partnerAddressCountry: "partnerAddress.partnerAddressCountry",

  partnerNationality: "partnerNationality.partnerNationality",

  partnerMaritalStatus: "partnerMaritalStatus.partnerMaritalStatus",

  dateOfMarriage: "marriageDetails.dateOfMarriage",
  placeOfMarriage: "marriageDetails.placeOfMarriage",

  deliveryFirstName: "deliveryDetails.deliveryFirstName",
  deliverySurname: "deliveryDetails.deliverySurname",
  deliveryAddressLine1: "deliveryDetails.deliveryAddressLine1",
  deliveryAddressLine2: "deliveryDetails.deliveryAddressLine2",
  deliveryCity: "deliveryDetails.deliveryCity",
  deliveryPostcode: "deliveryDetails.deliveryPostcode",
  deliveryPhone: "deliveryDetails.deliveryPhone",
};
