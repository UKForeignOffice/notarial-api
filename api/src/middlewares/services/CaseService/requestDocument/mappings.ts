export const order = {
  applicantDetails: "Applicant's details",
  applicantPassport: "Applicant's passport details",
  permanentAddress: "Applicant's permanent address",
  contactDetails: "Contact details",
  parent2ContactDetails: "Parent 2 contact details",
  reason: "Reason for document",
  certificate: "Certificate details",
  maritalStatus: "Marital status",
  marriageDetails: "Marriage details",
  partnerDetails: "Partner's details",
  delivery: "Delivery address",
  feedbackConsent: "Feedback consent",
};

export const remap = {
  applicationCountry: "country",
  post: "post",
  feedbackConsent: "feedbackConsent.feedbackConsent",

  firstName: "applicantDetails.firstName",
  middleName: "applicantDetails.middleName",
  surname: "applicantDetails.surname",

  passportDateOfIssue: "applicantPassport.dateOfIssue",
  passportNumber: "applicantPassport.number",
  dateOfBirth: "applicantPassport.dateOfBirth",
  countryOfBirth: "applicantPassport.countryOfBirth",
  placeOfBirth: "applicantPassport.placeOfBirth",

  "permanentAddress.addressLine1": "permanentAddress.addressLine1",
  "permanentAddress.addressLine2": "permanentAddress.addressLine2",
  "permanentAddress.city": "permanentAddress.city",
  "permanentAddress.postcode": "permanentAddress.postcode",
  "permanentAddress.country": "permanentAddress.country",

  docLanguage: "certificate.docLanguage",
  customLawCertType: "certificate.typeOfCertificate",

  mscReason: "reason.reason",
  reasonForLetter: "reason.reason",
  reasonForLetterOther: "reason.other",

  previouslyMarried: "maritalStatus.previouslyMarried",

  placeOfMarriage: "marriageDetails.placeOfMarriage",

  partnerIsBN: "partnerDetails.partnerIsBN",
  partnerFirstName: "partnerDetails.partnerFirstName",
  partnerMiddleName: "partnerDetails.partnerMiddleName",
  partnerSurname: "partnerDetails.partnerSurname",
  partnerPassportNumber: "partnerDetails.partnerPassportNumber",
  partnerDateOfBirth: "partnerDetails.partnerDateOfBirth",
  partnerCountryOfBirth: "partnerDetails.partnerCountryOfBirth",
  partnerPassportDateOfIssue: "partnerDetails.partnerPassportDateOfIssue",

  "delivery.country": "delivery.country",

  deliveryAddressLine1: "delivery.addressLine1",
  deliveryAddressLine2: "delivery.addressLine2",
  deliveryCity: "delivery.city",
  deliveryPostcode: "delivery.postCode",

  globalDeliveryAddressLine1: "delivery.addressLine1",
  globalDeliveryAddressLine2: "delivery.addressLine2",
  globalDeliveryCity: "delivery.city",
  globalDeliveryPostcode: "delivery.postCode",
  globalDeliveryCountry: "delivery.country",

  emailAddress: "contactDetails.emailAddress",
  phoneNumber: "contactDetails.phoneNumber",

  "parent2Contact.emailAddress": "parent2ContactDetails.emailAddress",
  "parent2Contact.phoneNumber": "parent2ContactDetails.phoneNumber",
};
