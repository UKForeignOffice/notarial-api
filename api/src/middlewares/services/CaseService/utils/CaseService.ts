import logger, { Logger } from "pino";
import { QueueService } from "../../QueueService";
import { FormType, PayMetadata } from "../../../../types/FormDataBody";
import config from "config";
import { getPostEmailAddress } from "../../utils/getPostEmailAddress";
import { PaymentViewModel, ProcessQueueData, ProcessQueueDataInput, SESSendJob } from "../types";
import { FormField } from "../../../../types/FormField";

export abstract class CaseService {
  logger: Logger;
  queueService: QueueService;
  templates: {
    SES: HandlebarsTemplateDelegate;
    Notify: Record<"postAlert", string>;
  };

  protected constructor({ queueService, templates }) {
    this.queueService = queueService;

    this.logger = logger().child({ service: "SES" });
    this.templates = templates;
  }

  abstract sendToProcessQueue(data: ProcessQueueData): Promise<string>;

  abstract sendEmail(data: ProcessQueueData): Promise<string>;

  /**
   * Builds the email body
   */
  abstract getEmailBody(data: { fields: FormField[]; payment?: PaymentViewModel; reference: string }, type: FormType): string;

  /**
   * builds the data required for the SES_PROCESS job
   */
  abstract buildProcessQueueData(input: ProcessQueueDataInput): ProcessQueueData;

  paymentViewModel(payment: PayMetadata | undefined, country: string) {
    if (!payment) {
      return;
    }
    const paymentUrl = new URL(payment.payId, config.get<string>("Pay.accountTransactionsUrl"));
    const allTransactionsByCountryUrl = new URL(config.get<string>("Pay.accountTransactionsUrl"));
    const total = payment.total ? (payment.total / 100).toFixed(2) : "Unpaid";
    allTransactionsByCountryUrl.searchParams.set("metadataValue", country);

    return {
      id: payment.payId,
      status: payment.state.status === "success" ? "success" : "cancelled or failed",
      url: paymentUrl.toString(),
      total,
      allTransactionsByCountry: {
        url: allTransactionsByCountryUrl.toString(),
        country,
      },
    };
  }

  /**
   * builds the data required for the SES_SEND job
   */
  abstract buildJobData(data: ProcessQueueData): SESSendJob;

  getPostAlertData(country: string, post: string, reference: string) {
    const emailAddress = getPostEmailAddress(post);
    if (!emailAddress) {
      this.logger.error(
        { code: "UNRECOGNISED_SERVICE_APPLICATION" },
        `No email address found for the specified post – ${country} - ${post} – reference ${reference}.`
      );
      return;
    }

    return {
      template: this.templates.Notify.postAlert,
      emailAddress,
      reference,
      options: {
        personalisation: {
          post,
          reference,
        },
        reference,
      },
    };
  }
}
