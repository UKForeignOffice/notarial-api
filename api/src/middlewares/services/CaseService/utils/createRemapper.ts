import { set } from "lodash";
import { FormField } from "../../../../types/FormField";

/**
 * Recreates an object with the keys remapped according to the mappings object.
 * The key is from the original object, the value is the new path in the new object.
 * If you need to ignore original keys or types, you can pass them in the ignoreKeys and ignoreTypes arrays.
 * This function is a curried function, pass in parameters to "set up" the remapper, then use the returned function to remap objects in a reducer function.
 *
 * @example
 * const remapper = createRemapper(affirmationMappings, ["ignoreThisKey"], ["ignoreThisType"]);
 * const remappedObject = remapper(formFields)
 */
export function createRemapper<T>(mappings: T, ignoreKeys: string[] = [], ignoreTypes: string[] = []) {
  const ignoreKeysSet = new Set(ignoreKeys);
  const ignoreTypesSet = new Set(ignoreTypes);
  return function (data: FormField[]) {
    return data.reduce((acc, curr) => {
      const key = curr.key;

      if (ignoreKeysSet.has(key) || ignoreTypesSet.has(curr.type)) {
        return acc;
      }

      const remappedPath = mappings[key];
      if (remappedPath) {
        set(acc, remappedPath, curr);
        return acc;
      }

      acc[key] = curr;

      return acc;
    }, {} as any);
  };
}
