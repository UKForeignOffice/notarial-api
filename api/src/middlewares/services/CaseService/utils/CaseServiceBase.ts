import { PayMetadata } from "../../../../types/FormDataBody";
import config from "config";
import { getPostEmailAddress } from "../../utils/getPostEmailAddress";
import logger, { Logger } from "pino";
import { CaseServiceBaseType } from "../types/CaseServiceBase";

export abstract class CaseServiceBase implements CaseServiceBaseType {
  logger: Logger;
  templates: {
    SES: HandlebarsTemplateDelegate;
    Notify: Record<"postAlert", string>;
  };

  protected constructor(sesTemplate: HandlebarsTemplateDelegate, postAlertTemplate: string) {
    this.logger = logger().child({ service: "SES" });
    this.templates = {
      SES: sesTemplate,
      Notify: {
        postAlert: postAlertTemplate,
      },
    };
  }
  paymentViewModel(payment: PayMetadata | undefined, country: string) {
    if (!payment) {
      return;
    }
    const paymentUrl = new URL(payment.payId, config.get<string>("Pay.accountTransactionsUrl"));
    const allTransactionsByCountryUrl = new URL(config.get<string>("Pay.accountTransactionsUrl"));
    const total = payment.total ? (payment.total / 100).toFixed(2) : "Unpaid";
    allTransactionsByCountryUrl.searchParams.set("metadataValue", country);

    return {
      id: payment.payId,
      status: payment.state.status === "success" ? "success" : "cancelled or failed",
      url: paymentUrl.toString(),
      total,
      allTransactionsByCountry: {
        url: allTransactionsByCountryUrl.toString(),
        country,
      },
    };
  }

  getPostAlertData(country: string, post: string, reference: string) {
    const emailAddress = getPostEmailAddress(post);
    if (!emailAddress) {
      this.logger.error(
        { code: "UNRECOGNISED_SERVICE_APPLICATION" },
        `No email address found for the specified post – ${country} - ${post} – reference ${reference}.`
      );
      return;
    }

    return {
      template: this.templates.Notify.postAlert,
      emailAddress,
      reference,
      options: {
        personalisation: {
          post,
          reference,
        },
        reference,
      },
    };
  }
}
