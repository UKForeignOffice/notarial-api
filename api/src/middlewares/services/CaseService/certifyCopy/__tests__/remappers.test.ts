import * as fields from "./fixtures/fields";
import { createRemapper } from "../../utils/createRemapper";
import { remap } from "../mappings";

/**
 * Tests to catch whether any remappers have changed.
 * check if `./fixtures/fields` should be updated with new or changed fields before updating the snapshots.
 */

test("certifyCopy is successfully remapped", () => {
  const remapped = createRemapper(remap)(fields.certifyCopy.remap);
  expect(remapped).toMatchSnapshot();
});
