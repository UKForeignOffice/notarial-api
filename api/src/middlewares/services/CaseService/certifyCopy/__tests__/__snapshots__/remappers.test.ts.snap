// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`certifyCopy is successfully remapped 1`] = `
{
  "applicantBirth": {
    "countryOfBirth": {
      "answer": "United Kingdom",
      "category": "applicantDetails",
      "key": "countryOfBirth",
      "title": "Country of birth",
      "type": "list",
    },
    "dateOfBirth": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "dateOfBirth",
      "title": "Date of birth",
      "type": "date",
    },
    "placeOfBirth": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "placeOfBirth",
      "title": "Place of birth ",
      "type": "text",
    },
  },
  "applicantDetails": {
    "firstName": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "firstName",
      "title": "First name",
      "type": "text",
    },
    "middleName": {
      "answer": null,
      "category": "applicantDetails",
      "key": "middleName",
      "title": "Middle name",
      "type": "text",
    },
    "sex": {
      "answer": "Male",
      "category": "applicantDetails",
      "key": "sex",
      "title": "Male or female",
      "type": "list",
    },
    "surname": {
      "answer": "test",
      "category": "applicantDetails",
      "key": "surname",
      "title": "Surname",
      "type": "text",
    },
  },
  "applicantPassport": {
    "answer": "https://document-upload-endpoint",
    "category": "documents",
    "dateOfIssue": {
      "answer": "1998-01-19",
      "category": "applicantDetails",
      "key": "passportDateOfIssue",
      "title": "Date of issue of passport",
      "type": "date",
    },
    "key": "applicantPassport",
    "number": {
      "answer": "12345678890",
      "category": "applicantDetails",
      "key": "passportNumber",
      "title": "Passport number",
      "type": "text",
    },
    "title": "Upload your UK passport",
    "type": "file",
  },
  "contactDetails": {
    "emailAddress": {
      "answer": "luke@cautionyourblast.com",
      "category": "contact",
      "key": "emailAddress",
      "title": "Confirm email address",
      "type": "text",
    },
    "phoneNumber": {
      "answer": "+44 1234 567809",
      "category": "contact",
      "key": "phoneNumber",
      "title": "Confirm phone number",
      "type": "text",
    },
  },
  "declaration": {
    "answer": true,
    "key": "declaration",
    "title": "Declaration",
    "type": "boolean",
  },
  "feedbackConsent": {
    "feedbackConsent": {
      "answer": false,
      "category": "feedback",
      "key": "feedbackConsent",
      "title": "Can our partner contact you for feedback to help improve this service?",
      "type": "list",
    },
  },
  "information": {
    "country": {
      "answer": "Vietnam",
      "category": "beforeYouStart",
      "key": "country",
      "title": "Application country",
      "type": "list",
    },
    "typeOfDocument": {
      "answer": "adult passport",
      "category": "beforeYouStart",
      "key": "typeOfDoc",
      "title": "Type of document to certify",
      "type": "list",
    },
  },
  "thirdParty": {
    "answer": false,
    "category": "beforeYouStart",
    "key": "thirdParty",
    "title": "Applying on behalf of someone else",
    "type": "list",
  },
}
`;
